### ISSUE TITLE
Support for cssClass attribute for sections in XML configs

### ISSUE THREAD

**Issue by stollr** [sulu/sulu#6908]
# Problem description

Currently the cssClass attribute is allowed in the xsd file for sections (see https://schemas.sulu.io/template/properties-1.0.xsd, line 74). But it is not supported by the code.

# Proposed solution

Add Propterty to SectionMetadata, adding the cssClass value in the PropertiesXmlParser and add support in the frontend.

**Comment by alexander-schranz** [comment]
@naitsirch this seems a left over of the 1.6 frontend. We are currently not planned to support this kind of things as it did make the old UI very inconsistent.

So I think we will in next major version remove this from the XSD file.

Do you had a specific usecase why you wanted to change a section?

**Comment by stollr** [comment]
Yes, indeed. I have extended the account details form xml file, included the origin xml via xinclude, and added my own section at the end.

Now the issue is that my own section starts at the left of the screen, whereas the origin form has two columns and the input fields are located in the right column.

I know that I could copy the origin xml and insert my custom fields in the right colum, but that would mean that I do not get updates of the form xml if you change something in the origin file.

I hope you can understand my issue.

By the way I have an off-topic question: Why haven't you used symfony forms for the backend and mapped the form views to react components?

This would decrease the learning curve for symfony devs ;-)

**Comment by alexander-schranz** [comment]
> Why haven't you used symfony forms for the backend and mapped the form views to react components?

Sulu Admin is by design Headless / API driven and Symfony Forms are not :) I understand the point that some would like to use Symfony Forms. But even Symfony Forms are not in all companys wide spreaded as they also have a certain complexity and are listed beside Security by Symfony itself as the complexes component.

React also was the choice as we also can implemented complex logic into our field types and make a things a lot more easier.

The usage of `XML` is more into the direction of page Templates and as we are CMS focused on Website Development. `XML` is very similar to `HTML` and most of the case `Frontend Developers` will Create Page Templates and as HTML and XML work the same way the learning curve for Frontend Developers is here very low and easy for them as they are already familiar with HTML. Where things like Symfony Form, even PHP or Yaml would be lot more complex for the Developer typically create the Website and so the Page XMLs.

For a typical Symfony Developer the usage of XMLs for Forms looks first a little bit strange, specially when going into creating custom Entity. But as said most usage of Forms are Page Templates and there we can make it via XMLs very easy for typical Frontend Developers which not even need to be familiar with PHP, Yaml, ... Still it is planned in future to support `PHP` or even `Yaml` as format for forms, but not Symfony Forms as we are still going over APIs and not server rendered HTML in the Admin.

Still we are happy here for any feedback if you are missing maybe a specific field types, because our target is mostly that 95% of the project should not require any custom JS admin code and Sulu should provide all required field types.

> I know that I could copy the origin xml and insert my custom fields in the right colum, but that would mean that I do not get updates of the form xml if you change something in the origin file.

I'm not sure but maybe creating a section inside your template with the same name could make them merged and so you only need your additional field in that section. Still the contact_details is very special. And our recommendation today is more about adding a `Tab` instead of editing or extending existing forms.

**Comment by stollr** [comment]
> Sulu Admin is by design Headless / API driven and Symfony Forms are not :)

What I meant was, configuring the forms with PHP and Symfony Forms and then only pass the configuration via API to the frontend and render the forms with react.

> The usage of XML is more into the direction of page Templates and as we are CMS focused on Website Development. XML is very similar to HTML [...] Where things like Symfony Form, even PHP or Yaml would be lot more complex for the Developer typically create the Website and so the Page XMLs.

XML for the page templates is fine. My question was only about forms for custom entities where Symfony controllers or the admin class has to be written in PHP anyways. And also the form type [configuration](https://docs.sulu.io/en/2.5/book/extend-admin.html#selection-field-type) could have been covered with symfony forms.

Back to the topic :-) your idea to use tabs is nice. I didn't thought about that and will give it a try.

Nonetheless maybe it would still be good to add something like "colOffset" in addition to "colSpan" to define an offset for the form columns. What do you think?
